src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const DAMAGE = 50;
+const DIED = 0;
+
 class Animal {
-  // write your code here
+  constructor(name, health = FULL_HEALTH) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden) {
+      prey.health -= DAMAGE;
+    }
+
+    if (prey.health === DIED) {
```

Comment:
The current implementation will only remove the animal from the 'alive' array when its health is exactly 0. If the health drops below 0, the animal will not be removed.